packages/web-pkg/src/components/FilesList/ResourceTiles.vue

```diff
@@ -1,46 +1,59 @@
-        <oc-list class="oc-tiles-sort-list">
-          <li v-for="(field, index) in sortFields" :key="index" class="oc-my-xs">
-            <oc-button
-              justify-content="space-between"
-              class="oc-tiles-sort-list-item oc-p-s oc-width-1-1"
-              :class="{
-                'oc-background-primary-gradient': isSortFieldSelected(field),
-                selected: isSortFieldSelected(field)
-              }"
-              :appearance="isSortFieldSelected(field) ? 'raw-inverse' : 'raw'"
-              :variation="isSortFieldSelected(field) ? 'primary' : 'passive'"
-              @click="selectSorting(field)"
-            >
-              <span v-text="$gettext(field.label)" />
-              <oc-icon v-if="isSortFieldSelected(field)" name="check" variation="inherit" />
-            </oc-button>
-          </li>
-        </oc-list>
-      </oc-drop>
+    <div class="oc-flex oc-flex-middle oc-mb-m oc-pb-s oc-tiles-controls">
+      <oc-checkbox
+        id="resource-table-select-all"
+        size="large"
+        :disabled="resources.length === disabledResources.length"
+        :label="$gettext('Select all')"
+        :label-hidden="true"
+        :model-value="areAllResourcesSelected"
+        @click.stop="toggleSelectionAll"
+      />
+      <div v-if="sortFields.length" class="oc-tile-sorting oc-ml-s">
+        <span class="oc-mr-xs" v-text="$gettext('Sort by: ')" />
+        <oc-button id="oc-tiles-sort-btn" appearance="raw" gap-size="none">
+          <span v-text="$gettext(currentSortField.label)" />
+          <oc-icon name="arrow-down-s" />
+        </oc-button>
+        <oc-drop
+          ref="sortDrop"
+          toggle="#oc-tiles-sort-btn"
+          class="oc-tiles-sort-drop"
+          mode="click"
+          padding-size="small"
+          close-on-click
+        >
+          <oc-list class="oc-tiles-sort-list">
+            <li v-for="(field, index) in sortFields" :key="index" class="oc-my-xs">
+              <oc-button
+                justify-content="space-between"
+                class="oc-tiles-sort-list-item oc-p-s oc-width-1-1"
+                :class="{
+                  'oc-background-primary-gradient': isSortFieldSelected(field),
+                  selected: isSortFieldSelected(field)
+                }"
+                :appearance="isSortFieldSelected(field) ? 'raw-inverse' : 'raw'"
+                :variation="isSortFieldSelected(field) ? 'primary' : 'passive'"
+                @click="selectSorting(field)"
+              >
+                <span v-text="$gettext(field.label)" />
+                <oc-icon v-if="isSortFieldSelected(field)" name="check" variation="inherit" />
+              </oc-button>
+            </li>
+          </oc-list>
+        </oc-drop>
+      </div>
     </div>
     <oc-list class="oc-tiles oc-flex">
       <li
         v-for="resource in resources"
         :key="resource.id"
         class="oc-tiles-item has-item-context-menu"
+        v-memo="[resources]"
```

Comment:
@JammingBen @kulmann I added the v-memo directive, so the list gets only re-rendered when the resources do change (which is not the case when a resource gets selected) which makes the the selection quite fast with 500 tiles and fixes https://github.com/owncloud/web/issues/10578.

I couldn't spot any bad side effects

Reply:
Nice! Can/Should we use it for the files table as well? 🤔

Reply:
From my observations, the table doesn't get rerendered again when props change but the list in the resource tiles component did, so I don't think it's necessary

Reply:
> Something messes with the right-click on tiles, it only works the first time for me.

Aaaaaaaand that is the v-memo side effect urghs

Reply:
Removed the v-memo, this is not working correctly and scope creep, need to investigate further 🕵️